core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -4761,7 +4761,7 @@ protected function parentView()
 	protected function listView()
 	{
 		$table = ($GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['mode'] ?? null) == self::MODE_TREE_EXTENDED ? $this->ptable : $this->strTable;
-		$orderBy = $GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['fields'] ?? array();
+		$orderBy = $GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['fields'] ?? array('id');
 		$firstOrderBy = preg_replace('/\s+.*$/', '', $orderBy[0]);
```

Comment:
Shouldn’t we fix this on the second line instead?

```suggestion
		$orderBy = $GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['fields'] ?? array();
		$firstOrderBy = preg_replace('/\s+.*$/', '', $orderBy[0] ?? '');
```

Reply:
I thought about this as well, but ordering by an empty string is imho worse than ordering by the record id (maybe sorting by empty won't even work).

Reply:
I’m not unserstanding all the rest of the code, is it guaranteed that there is always a column named `'id'`?
If so, using `array('id')` might be better I agree.

Reply:
well it's a DCA, and our DCA does not work without an ID column imho 🤷